integration/storagetest/maptestrunner.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storagetest
+
+import (
+	"context"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+	"testing"
+
+	"github.com/google/trillian/storage"
+)
+
+type MapStorageFactory func(ctx context.Context, t *testing.T) (storage.MapStorage, storage.AdminStorage)
+type MapStorageTest func(ctx context.Context, t *testing.T, s storage.MapStorage, as storage.AdminStorage)
+
+// RunMapStorageTests runs all the map storage tests against the provided map storage implementation.
+func RunMapStorageTests(t *testing.T, storageFactory MapStorageFactory) {
+	ctx := context.Background()
+	for _, f := range mapTestFunctions(t, &MapTests{}) {
+		ms, as := storageFactory(ctx, t)
+		t.Run(functionName(f), func(t *testing.T) { f(ctx, t, ms, as) })
+	}
+}
+
+func functionName(x interface{}) string {
```

Comment:
You could write a short doc comment here with an example to show what is returned.